packages/block-editor/src/components/aspect-ratio-dropdown/index.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * WordPress dependencies
+ */
+import { check, aspectRatio as aspectRatioIcon } from '@wordpress/icons';
+import { DropdownMenu, MenuGroup, MenuItem } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useSettings } from '../use-settings';
+import { POPOVER_PROPS } from '../image-editor/constants';
+import { useImageEditingContext } from '../image-editor/context';
+
+function AspectRatioGroup( {
+	aspectRatios,
+	isDisabled,
+	label,
+	onClick,
+	value,
+} ) {
+	return (
+		<MenuGroup label={ label }>
+			{ aspectRatios.map( ( { name, slug, ratio } ) => (
+				<MenuItem
+					key={ slug }
+					disabled={ isDisabled }
+					onClick={ () => {
+						onClick( ratio );
+					} }
+					role="menuitemradio"
+					isSelected={ ratio === value }
+					icon={ ratio === value ? check : undefined }
+				>
+					{ name }
+				</MenuItem>
+			) ) }
+		</MenuGroup>
+	);
+}
+
+function ratioToNumber( str ) {
+	// TODO: support two-value aspect ratio?
+	// https://css-tricks.com/almanac/properties/a/aspect-ratio/#aa-it-can-take-two-values
+	const [ a, b, ...rest ] = str.split( '/' ).map( Number );
+	if (
+		a <= 0 ||
+		b <= 0 ||
+		Number.isNaN( a ) ||
+		Number.isNaN( b ) ||
+		rest.length
+	) {
+		return NaN;
+	}
+	return b ? a / b : a;
+}
+
+function presetRatioAsNumber( { ratio, ...rest } ) {
+	return {
+		ratio: ratioToNumber( ratio ),
+		...rest,
+	};
+}
+
+export default function AspectRatioDropdown( { toggleProps } ) {
+	const { isInProgress, aspect, setAspect, defaultAspect } =
+		useImageEditingContext();
+
+	const [ defaultRatios ] = useSettings( 'dimensions.aspectRatios.default' );
+	const [ themeRatios ] = useSettings( 'dimensions.aspectRatios.theme' );
+	const [ showDefaultRatios ] = useSettings(
+		'dimensions.defaultAspectRatios'
+	);
```

Comment:
Let's combine the setting getters - `useSettings( 'dimensions.aspectRatios.default', 'dimensions.aspectRatios.theme', 'dimensions.defaultAspectRatios' )`

Related: https://make.wordpress.org/core/2024/03/09/miscellaneous-editor-changes-in-wordpress-6-5/#new-usesettings-hook-for-reading-block-instance-settings

Reply:
Ahh very good call 🤦